File: algorithms/src/main/scala/graphs/leejava/LeeApp.java
package graphs.leejava;

import java.util.Optional;
import java.util.Set;

public class LeeApp {

  public static void main(String[] args) {
    Lee lee = new Lee(20, 15);
    Point src = Point.of(0, 0);
    Point dst =
//      Point.of(19, 14);
      Point.of(19, 0);
    Set<Point> obstacles = Set.of(
      Point.of(5,14),
      Point.of(5,13),
      Point.of(5,12),
      Point.of(5,11),
      Point.of(5,10),
      Point.of(5,9),
      Point.of(5,8),

      Point.of(10,0),
      Point.of(10,1),
      Point.of(10,2),
      Point.of(10,3),
      Point.of(10,4),
      Point.of(10,5),
      Point.of(10,6),
      Point.of(10,7),
      Point.of(10,8)
    );

    System.out.println(1);
    Optional<Iterable<Point>> trace = lee.trace(src, dst, obstacles);
    System.out.println(2);
    System.out.println(trace);
    System.out.println();
    System.out.println(lee);
    System.out.println();
    trace.ifPresent(path -> System.out.println(lee.boardFormatted(path)));
  }

}

File: algorithms/src/main/scala/graphs/lee/LeeApp.scala
package graphs.lee

object LeeApp extends App {

  val src       = Point(0, 0)
  val dst       = Point(19, 0)
  val obstacles = Set(
    (5, 14),
    (5, 13),
    (5, 12),
    (5, 11),
    (5, 10),
    (5, 9),
    (5, 8),
    (10, 0),
    (10, 1),
    (10, 2),
    (10, 3),
    (10, 4),
    (10, 5),
    (10, 6),
    (10, 7),
    (10, 8)
  ).map { case (x, y) => Point(x, y) }
  val lee       = new Lee(20, 15, obstacles)
  val trace     = lee.trace(src, dst)
  System.out.println(trace)
  System.out.println()
  trace.foreach(path => System.out.println(lee.fmtBoard(path)))

}
